databricks_cli/secrets/cli.py

```diff
@@ -0,0 +1,308 @@
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Writes an access control rule for a principal applied to '
+                          ' a given secret scope.')
+@click.option('--scope', required=True)
+@click.option('--principal', required=True)
+@click.option('--permission', type=click.Choice(['MANAGE', 'WRITE', 'READ']),
+              required=True)
+@profile_option
+@eat_exceptions
+@provide_api_client
+def write_acl(api_client, scope, principal, permission):
+    """
+    Creates or overwrites the ACL associated with the given principal (user or group) on the
+    specified secret scope.
+    """
+    SecretApi(api_client).write_acl(scope, principal, permission)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Deletes an access control rule for a principal.')
+@click.option('--scope', required=True)
+@click.option('--principal', required=True)
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_acl(api_client, scope, principal):
+    """
+    Deletes the given ACL on the given secret scope.
+    """
+    SecretApi(api_client).delete_acl(scope, principal)
+
+
+def _acls_to_table(acls_json):
+    ret = []
+    for s in acls_json.get('items', []):
+        ret.append((s['principal'], s['permission'].upper()))
+    return ret
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Lists all access control rules for a given secret scope.')
+@click.option('--scope', required=True)
+@click.option('--output', default=None, help=OutputClickType.help, type=OutputClickType())
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_acls(api_client, scope, output):
+    """
+    Lists the ACLs set on the given secret scope.
+    """
+    acls_json = SecretApi(api_client).list_acls(scope)
+    if OutputClickType.is_json(output):
+        click.echo(pretty_format(acls_json))
+    else:
+        click.echo(tabulate(_acls_to_table(acls_json), headers=ACL_HEADER))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Gets the details for an access control rule.')
+@click.option('--scope', required=True)
+@click.option('--principal', required=True)
+@click.option('--output', default=None, help=OutputClickType.help, type=OutputClickType())
+@profile_option
+@eat_exceptions
+@provide_api_client
+def get_acl(api_client, scope, principal, output):
+    """
+    Describes the details about the given ACL for the principal and secret scope.
+    """
+    acl_json = SecretApi(api_client).get_acl(scope, principal)
+    if OutputClickType.is_json(output):
+        click.echo(pretty_format(acl_json))
+    else:
+        acl_list = _acls_to_table({'items': [acl_json]})
+        click.echo(tabulate(acl_list, headers=ACL_HEADER))
+
+
+@click.group(context_settings=CONTEXT_SETTINGS,
+             short_help='Utility to interact with Databricks secret API.'
+                        ' (Available only in Azure Databricks)')
```

Comment:
Remove the only available in Azure Databricks, so we don't need to make any updates to the CLI when we release AWS. The cloud-specific error messaging is handled by the ApiProxy now.